### ISSUE TITLE
[S3] :DeleteObjects Missing required header Content-MD5

### ISSUE THREAD

**Issue by RickMoynihan** [cognitect-labs/aws-api#40]
When calling :DeleteObjects against S3:

```
(aws/invoke client {:op :DeleteObjects, :request {:Delete {:Objects [{:Key "annual/my-file"}]}, :Bucket "test-bucket"}}
```

I get the following error: 

```
1. Unhandled clojure.lang.ExceptionInfo
   Error invoking AWS service
   {:Error
    {:Code "InvalidRequest",
     :Message "Missing required header for this request: Content-MD5",
     :RequestId "63DDF59BD7350E9C",
     :HostId
     "BjCzOVMPQIw912la2lnv/FuJE64YeP2nbFtwx58ei9KKFEfIa2kv96JkqAS0vIdw6LaT3anuLAk="},
    :cognitect.anomalies/category :cognitect.anomalies/incorrect}

                    s3.clj:   36  data-loader.concerns.s3/invoke
                    s3.clj:   31  data-loader.concerns.s3/invoke
                      REPL:   87  data-loader.pipelines.annual.annual-test/eval33563
                      REPL:   85  data-loader.pipelines.annual.annual-test/eval33563
             Compiler.java: 7176  clojure.lang.Compiler/eval
             Compiler.java: 7131  clojure.lang.Compiler/eval
                  core.clj: 3214  clojure.core/eval
                  core.clj: 3210  clojure.core/eval
                  main.clj:  414  clojure.main/repl/read-eval-print/fn
                  main.clj:  414  clojure.main/repl/read-eval-print
                  main.clj:  435  clojure.main/repl/fn
                  main.clj:  435  clojure.main/repl
                  main.clj:  345  clojure.main/repl
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  core.clj:  665  clojure.core/apply
                  core.clj:  660  clojure.core/apply
                regrow.clj:   18  refactor-nrepl.ns.slam.hound.regrow/wrap-clojure-repl/fn
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   83  nrepl.middleware.interruptible-eval/evaluate/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  665  clojure.core/apply
                  core.clj: 1973  clojure.core/with-bindings*
                  core.clj: 1973  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   81  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   50  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  221  nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
    interruptible_eval.clj:  189  nrepl.middleware.interruptible-eval/run-next/fn
                  AFn.java:   22  clojure.lang.AFn/run
   ThreadPoolExecutor.java: 1149  java.util.concurrent.ThreadPoolExecutor/runWorker
   ThreadPoolExecutor.java:  624  java.util.concurrent.ThreadPoolExecutor$Worker/run
               Thread.java:  748  java.lang.Thread/run
```

DeleteObject and ListBucket both work with my permission set.   `aws/validate-requests` does not report a problem with the request.

**Comment by dchelimsky** [comment]
Thanks for the report. This is one of those special cases in which the AWS documentation provides more information than the service descriptions do. I've got a fix locally - just trying to figure out how to decide when to include this header.

**Comment by stuarthalloway** [comment]
Example of how Java does it: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/AddContentMd5HeaderInterceptor.java

**Comment by dchelimsky** [comment]
Fixed by cc068de9e732291802e7b5d506685525fbc4dda3

**Comment by dchelimsky** [comment]
Released in 0.8.266
